packages/react-ui/src/widgets/FeatureSelectionWidgetUI.js

```diff
@@ -0,0 +1,364 @@
+        activated={activated}
+      >
+        <SelectedModeViewer
+          modes={[...drawModes, ...editModes.map((mode) => ({ ...mode, isEdit: true }))]}
+          selectedMode={selectedMode}
+          activated={activated}
+          onActivatedChange={onActivatedChange}
+          tooltipPlacement={tooltipPlacement}
+        />
+      </Helper>
+      <ModesSelector
+        drawModes={drawModes}
+        editModes={editModes}
+        selectedMode={selectedMode}
+        onSelectMode={onSelectMode}
+        activated={activated}
+        tooltipPlacement={tooltipPlacement}
+      />
+      {!!geometries?.length && (
+        <GeometriesViewer
+          geometries={geometries}
+          onSelectGeometry={onSelectGeometry}
+          onDeleteGeometry={onDeleteGeometry}
+          tooltipPlacement={tooltipPlacement}
+        />
+      )}
+    </Wrapper>
+  );
+}
+
+FeatureSelectionWidgetUI.defaultProps = {
+  className: '',
+  activated: false,
+  geometries: [],
+  tooltipPlacement: 'bottom',
+  editModes: []
+};
+
+const MODE_SHAPE = PropTypes.shape({
+  id: PropTypes.string.isRequired,
+  label: PropTypes.string.isRequired,
+  icon: PropTypes.element.isRequired
+});
+
+FeatureSelectionWidgetUI.propTypes = {
+  className: PropTypes.string,
+  drawModes: PropTypes.arrayOf(MODE_SHAPE.isRequired).isRequired,
+  editModes: PropTypes.arrayOf(MODE_SHAPE.isRequired),
+  selectedMode: PropTypes.string.isRequired,
+  onSelectMode: PropTypes.func,
+  activated: PropTypes.bool,
+  onActivatedChange: PropTypes.func,
+  geometries: PropTypes.array,
+  onSelectGeometry: PropTypes.func,
+  tooltipPlacement: PropTypes.string
+};
+
+export default FeatureSelectionWidgetUI;
+
+// Aux
+function Helper({ hasMode, isEdit, activated, children }) {
+  const alreadyOpenedRef = useRef({});
+  const shouldOpen = hasMode && activated;
+
+  const [open, setOpen] = useState(shouldOpen);
+
+  useEffect(() => {
+    const wasAlreadyOpened = alreadyOpenedRef.current[isEdit];
+    if (wasAlreadyOpened) {
+      if (open) {
+        setOpen(false);
+      }
+      return;
+    }
+
+    if (shouldOpen) {
+      alreadyOpenedRef.current = {
+        ...alreadyOpenedRef.current,
+        [isEdit]: true
+      };
+      setOpen(true);
+      let timeout = setTimeout(() => setOpen(false), 5000);
+
+      return () => clearTimeout(timeout);
+    }
+    // Logic around show/hide helper tooltip is manage by shouldOpen var
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [shouldOpen]);
+
+  return (
+    <Tooltip
+      title={
+        isEdit
+          ? 'Select geometry to move or edit it'
+          : 'Click on the map to define your spatial filter'
+      }
+      open={open}
+    >
+      <Box>{children}</Box>
+    </Tooltip>
+  );
+}
+
+const useGeometriesViewerStyles = makeStyles((theme) => ({
+  chip: {
+    margin: theme.spacing(0, 0, 0, 1)
+  }
+}));
+
+function GeometriesViewer({
+  geometries,
```

Comment:
Overfeature here. We don't need to support an array of geometries. With only 1 geometry is enough for this scope

Reply:
Jose and I found interesting to have this feature in mind for a future implementation. Currently, we don't support multiple geometries, but UI can be used without Widget, and I think it can be useful for developers. In any case, I'll remove it.

Reply:
Keep it simple and add it in a feature implementation

Reply:
In a future implementation, changing geometry to geometries will be a breaking change. The code won't be more simple just removing a map. But ok.

Reply:
Keep it simple, we'll make the code more complex at the time is required, not before. it adds complexity in the meantime